cpp/src/arrow/compute/kernels/vector_selection_test.cc

```diff
@@ -2365,38 +2365,91 @@ TEST_F(TestDropNullKernelWithTable, DropNullTableWithSlices) {
   });
 }
 
-TEST(TestIndicesNonZero, IndicesNonZero) {
+template <typename Type>
+class TestIndicesNonZero : public ::testing::Test {};
+
+TYPED_TEST_SUITE(TestIndicesNonZero, NumericArrowTypes);
```

Comment:
This will compile a distinct version of the tests for each tested type, but this doesn't seem necessary.
How about instead:
```c++
TEST(TestIndicesNonZero, IndicesNonZero) {
  Datum actual;
  std::shared_ptr<Array> result;
  for (const auto& type : NumericTypes()) {
    ARROW_SCOPED_TRACE("Input type = ", type->ToString());
    ASSERT_OK_AND_ASSIGN(actual,  // etc.
```

Reply:
done